src/TestFramework/CommandLineBuilder.php

```diff
@@ -76,8 +77,13 @@ public function build(string $testFrameworkExecutable, array $phpExtraArgs, arra
          *
          * This lets folks use, say, a bash wrapper over phpunit.
          */
-        if ('cli' === PHP_SAPI && $phpExtraArgs === [] && is_executable($testFrameworkExecutable) && shell_exec('command -v php') !== null) {
-            return array_merge([$testFrameworkExecutable], $frameworkArgs);
+        if ('cli' === PHP_SAPI && $phpExtraArgs === [] && is_executable($testFrameworkExecutable)) {
+            try {
+                shell_exec('command -v php');
+
+                return array_merge([$testFrameworkExecutable], $frameworkArgs);
```

Comment:
Alternatively we can use `unsafe_shell_exec` here.
```
use function shell_exec as unsafe_shell_exec;
```
I don't remember what kind of hoops and loops required to make it pass the CI, but if it improves the readability...

Reply:
I reverted this and added a suppression to PHPStan.